src/main/utilities/src/Polygon.cpp

```diff
@@ -14,16 +14,37 @@
 using std::cout;
 using std::endl;
 
-Polygon::Polygon(GLint pointCount, GLuint programId, vector<GLfloat> vertices, vector<GLfloat> textures,
-    vector<GLfloat> normals) : Polygon(pointCount, programId, vertices) {
-    cout << "Creating GameObject" << endl;
+static int polyCount;
 
+/**
+ * @brief Constructs a new Polygon, adds texture coordinates/normal vectors
+ * 
+ * @param triCount Number of triangles in the shape
+ * @param programId ProgramId used to identify the shader set associated with this polygon
+ * @param vertices Vertex points for triangles making up the polygon
+ * @param textures Texture coordinates for the polygon
+ * @param normals Normal vector for the polygon tri faces
+ */
+Polygon::Polygon(GLuint triCount, GLuint programId, vector<GLfloat> vertices, vector<GLfloat> textures,
```

Comment:
The Polygon class shouldn't care about the programId either. Polygon shouldn't need to interact with the GfxController PERIOD. It should just hold data for generated objects.